com.sun.xml.bind's module descriptor missing `provides`

I download the JAXB RI 2.4.0 and tried to use the java.xml.bind module on the module path. Alas, the implementation module is not resolved, it seems to be missing `provides javax.xml.bind.JAXBContextFactory` from its declaration so it will not be resolved.

$ javac -p mod --add-modules java.xml.bind SimpleTest.java
$ java -p mod --add-modules java.xml.bind SimpleTest
javax.xml.bind.JAXBException: Implementation of JAXB-API has not been 
found on module path or classpath.
 - with linked exception:
[java.lang.ClassNotFoundException: com.sun.xml.bind.v2.ContextFactory]
    at 
java.xml.bind/javax.xml.bind.ContextFinder.newInstance(ContextFinder.java:269)
    at 
java.xml.bind/javax.xml.bind.ContextFinder.find(ContextFinder.java:412)
    at 
java.xml.bind/javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:721)
    at 
java.xml.bind/javax.xml.bind.JAXBContext.newInstance(JAXBContext.java:662)
    at JAXBExample.main(JAXBExample.java:17)
Caused by: java.lang.ClassNotFoundException: 
com.sun.xml.bind.v2.ContextFactory
    at 
java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:583)
    at 
java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
    at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
    at 
java.xml.bind/javax.xml.bind.ServiceLoaderUtil.nullSafeLoadClass(ServiceLoaderUtil.java:122)
    at 
java.xml.bind/javax.xml.bind.ServiceLoaderUtil.safeLoadClass(ServiceLoaderUtil.java:155)
    at 
java.xml.bind/javax.xml.bind.ContextFinder.newInstance(ContextFinder.java:267)
    ... 4 more

It can be worked around by specifying `--add-modules com.sun.xml.bind` but this should not be necessary.

--- a/jaxb-ri/runtime/impl/src/main/java/com/sun/xml/bind/v2/JAXBContextFactory.java
+++ b/jaxb-ri/runtime/impl/src/main/java/com/sun/xml/bind/v2/JAXBContextFactory.java
@@ -1,0 +1,40 @@
+package com.sun.xml.bind.v2;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import java.util.Map;
+
+/**
+ * Creates JAXB context.
+ * Context factory supporting JavaSE service loading facilities.
+ *
+ * Redirects to {@link ContextFactory}, which is not removed due to compatibility reasons.
+ */
+public class JAXBContextFactory implements javax.xml.bind.JAXBContextFactory {
+
+    /**
+     * Creates JAXB context.
+     *
+     * @param classesToBeBound JAXB classes accessed by runtime.
+     * @param properties JAXB properties passed to runtime.
+     * @return JAXB context.
+     * @throws JAXBException on any error.
+     */
+    @Override
+    public JAXBContext createContext(Class<?>[] classesToBeBound, Map<String, ?> properties) throws JAXBException {
+        return ContextFactory.createContext(classesToBeBound, (Map<String, Object>) properties);
+    }
+
+    /**
+     * Creates JAXB context.
+     * @param contextPath path to scan for JAXB classes accessed by runtime.
+     * @param classLoader class loader to use.
+     * @param properties JAXB properties.
+     * @return JAXB context.
+     * @throws JAXBException on any error.
+     */
+    @Override
+    public JAXBContext createContext(String contextPath, ClassLoader classLoader, Map<String, ?> properties) throws JAXBException {
+        return ContextFactory.createContext(contextPath, classLoader, (Map<String, Object>) properties);
+    }
+}

--- a/jaxb-ri/runtime/impl/src/main/java/module-info.java
+++ b/jaxb-ri/runtime/impl/src/main/java/module-info.java
@@ -52,4 +52,6 @@ module com.sun.xml.bind {
     opens com.sun.xml.bind.v2.model.nav to com.sun.tools.xjc;
 
     uses javax.xml.bind.JAXBContextFactory;
+
+    provides javax.xml.bind.JAXBContextFactory with com.sun.xml.bind.v2.JAXBContextFactory;
 }

